modules/drivers/mongo/test/metabase/driver/mongo_test.clj

```diff
@@ -197,36 +196,44 @@
              {:schema nil, :name "reviews"}}
             (:tables (driver/describe-database :mongo (mt/db)))))))
 
-(deftest ^:parallel describe-table-test
+(defmacro with-describe-table-query-depths
+  "Test various values of `describe-table-query-depth` to prove that fields are synced the same regardless of the level of nesting."
+  [& body]
+  `(doseq [depth# [0 5]]
+     (with-redefs [mongo/describe-table-query-depth depth#]
+       ~@body)))
+
+(deftest describe-table-test
```

Comment:
Would it be worth it to add a test just for `describe-table-query`, so given some certain input we can check what sort of native query we expect it to generate?